src/car/Car.java

```diff
@@ -0,0 +1,79 @@
+package car;
+
+import java.io.*;
+
+public class Car implements Cloneable, Serializable {
+    private String model;
+    private String color;
+    private Engine engine;
+
+    public Car(String model, String color, Engine engine) {
+        this.model = model;
+        this.color = color;
+        this.engine = engine;
+    }
+
+    public String getModel() {
+        return model;
+    }
+
+    public void setModel(String model) {
+        this.model = model;
+    }
+
+    public String getColor() {
+        return color;
+    }
+
+    public void setColor(String color) {
+        this.color = color;
+    }
+
+    public Engine getEngine() {
+        return engine;
+    }
+
+    public void setEngine(Engine engine) {
+        this.engine = engine;
+    }
+
+    @Override
+    public Car clone() {
+        try {
+            return (Car) super.clone();
+        } catch (CloneNotSupportedException e) {
+            return new Car(this.model, this.color, this.engine);
+        }
+    }
+
+    public Object deepClone() {
+        ByteArrayInputStream byteArrayInputStream;
+        ObjectInputStream objectInputStream;
+        Object cloneObject = null;
+
+        try {
+            ByteArrayOutputStream byteArrayOutputStream = new ByteArrayOutputStream();
+            ObjectOutputStream objectOutputStream = new ObjectOutputStream(byteArrayOutputStream);
+
+            objectOutputStream.writeObject(this);
+            objectOutputStream.close();
```

Comment:
better close recourse in the finally block.

Reply:
I used "try with resources" before and forgot to change this block.

Reply:
you can use try with recourse too